kong/api/routes/config.lua

```diff
@@ -97,14 +97,26 @@ return {
         return declarative.load_into_cache_with_events(entities, meta, new_hash)
       end)
 
-      if err == "no memory" then
-        kong.log.err("not enough cache space for declarative config")
-        return kong.response.exit(413, {
-          message = "Configuration does not fit in Kong cache"
-        })
-      end
-
       if not ok then
+        if err == "busy" or err == "locked" then
+          return kong.response.exit(429, {
+            message = "Currently loading previous configuration."
+          }, { ["Retry-After"] = 60 })
```

Comment:
Can we somehow get `ttl` of the lock, and say set this to time when the lock will timeout, or is it possible that lock is held forever?

Reply:
We can leave this later.